docs/spec.md

```diff
@@ -2633,6 +2633,54 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
+
+Fills a `result` tensor with values in increasing order starting from zero
+along the `iota_dimension` dimension. `result` shape is provided
+dynamically via `output_shape`.
+
+More formally, for all `result_index` in `index_space(output_shape)`,
```

Comment:
+1 to @ghpvnist comment on ensure all criteria in [spec_checklist.md](https://github.com/openxla/stablehlo/blob/main/docs/spec_checklist.md) is met.